cmd/root.go

```diff
@@ -220,3 +222,45 @@ func prepConfig() error {
 	}
 	return nil
 }
+
+func exportTransformers() []transformer.StorageTransformerInitializer {
+	prepConfig()
+
+	// Get the plugin path and load the plugin
+	_, pluginPath, pathErr := genConfig.GetPluginPaths()
+
+	if pathErr != nil {
+		LogWithCommand.Fatalf("failed to get plugin paths: %s", pathErr.Error())
+	}
+
+	LogWithCommand.Info("linking plugin ", pluginPath)
+	plug, openErr := plugin.Open(pluginPath)
+	if openErr != nil {
+		LogWithCommand.Fatalf("linking plugin failed: %s", openErr.Error())
+	}
+
+	// Load the `Exporter` symbol from the plugin
+	LogWithCommand.Info("loading transformers from plugin")
+	symExporter, lookupErr := plug.Lookup("Exporter")
+	if lookupErr != nil {
+		LogWithCommand.Fatalf("loading Exporter symbol failed: %s", lookupErr.Error())
+	}
+
+	// Assert that the symbol is of type Exporter
+	exporter, ok := symExporter.(Exporter)
+	if !ok {
+		LogWithCommand.Fatal("plugged-in symbol not of type Exporter")
```

Comment:
nbd but I'd be a fan of returning an error and going the `Fatal` logging from the caller - don't think that has a material consequence here but prob a good habit in case there are surrounding processes that need to shutdown gracefully

Reply:
That makes sense to me! I've updated `getStorageValue` command to follow that pattern, but decided to leave `execute` as is for now, and just do a `LogWithCommand.Fatal`. `execute` would require the underlying method to be refactored to return an error, and I am a little apprehensive in dealing with (and it's waitgroups 😱 ) as part of this change since it's already feeling kind of big.